src/main/antlr4/Manifold.g4

```diff
@@ -39,25 +44,53 @@ namespacedIdentifier: (IDENTIFIER '::')* IDENTIFIER;
 //                                                    //
 ////////////////////////////////////////////////////////
 
-typevalue:
+type:
     namespacedIdentifier # Typename
   | tupleTypeValue # TupleType
+  ;
+
+typevalue:
+    type # SimpleType
   | functionTypeValue # FunctionType
   ;
 
-expression:
+// TODO: implement type declarations and undefined variable declarations
+//declaration:
+//    type namespacedIdentifier # UndefinedDeclaration
+//  | TYPE_KEYWORD namespacedIdentifier '=' type # TypeDeclaration
+//  ;
+
+reference:
+    tupleValue # Tuple
+  | reference '.' IDENTIFIER # StaticAttributeAccessExpression
+  | reference '[' INTEGER_VALUE ']' # StaticAttributeAccessExpression
+  | namespacedIdentifier # VariableReferenceExpression
+  ;
+
+// TODO implement arglist for multiple parameter passing to functions
+// arglist:
+//     rvalue arglist | ;
+
+rvalue:
```

Comment:
Is there a reason why we cannot have the rule `rvalue : lvalue` to avoid repeating ourselves?